src/App/mermaidData/syncApiDataIntoOfflineStorage/useInitializeSyncApiDataIntoOfflineStorage.js

```diff
@@ -2,19 +2,10 @@ import { useEffect, useRef } from 'react'
 import { useLocation } from 'react-router-dom'
 import { toast } from 'react-toastify'
 import language from '../../../language'
+import { getProjectIdFromLocation } from '../../../library/getProjectIdFromLocation'
 import { getToastArguments } from '../../../library/getToastArguments'
 import { useSyncStatus } from './SyncStatusContext'
 
-const getProjectIdFromLocation = (location) => {
-  const { pathname } = location
-
-  const pathNameParts = pathname.split('/')
-
-  const projectId = pathNameParts[pathNameParts.indexOf('projects') + 1]
-
-  return projectId
-}
-
```

Comment:
eek, this logic was faulty because index of projects might in the future be -1 (eg if we add a new route for `/documentation` or anything), and then because -1 + 1 = 0 projectId would end up being `/documentation` and it would cause errors everywhere with syncing probably. Bad historical Melissa.